packages/nextjs/src/config/webpack.ts

```diff
@@ -43,7 +43,7 @@ export function constructWebpackConfigFunction(
   // Will be called by nextjs and passed its default webpack configuration and context data about the build (whether
   // we're building server or client, whether we're in dev, what version of webpack we're using, etc). Note that
   // `incomingConfig` and `buildContext` are referred to as `config` and `options` in the nextjs docs.
-  const newWebpackFunction = (incomingConfig: WebpackConfigObject, buildContext: BuildContext): WebpackConfigObject => {
+  return (incomingConfig: WebpackConfigObject, buildContext: BuildContext): WebpackConfigObject => {
```

Comment:
If we're going to do this, can we still name it? (It makes debugging easier.)

```suggestion
  return function newWebpackFunction(incomingConfig: WebpackConfigObject, buildContext: BuildContext): WebpackConfigObject {
```

Reply:
Sure: cd94b1257ac0998c35a28d06bcbb6a0f26a1c27e

The reason I made this change was because it made sonarlint super angry and I was tired of looking at it :p. Screenshot for reference:

![Screen Shot 2022-08-10 at 14 19 54](https://user-images.githubusercontent.com/8118419/183899424-ae9a3fc4-8326-44d7-b736-eb5d74e21a99.png)

I guess it has a point since when you're reading the `constructWebpackConfigFunction` from top to bottom you now instantly know that `newWebpackFunction` will be returned, whereas before you would have to look all the way to the bottom.

(Wanna sneak in the thing Abhy mentioned the other day about `function` vs. `() => {}` and I generally agree that we should keep the situations where we use `functions` to only those situations where it makes a semantic difference. It will help us differentiate the situations where we have to care about `this` vs the situations where we don't have to care about it.)

Reply:
Ah, yes - I have that rule turned off in my sonarlint (because it kept yelling at me!). It's mad because you don't _have to_ assign to a variable, which is true. But in a situation where a function is returning a big long thing, I nonetheless generally _choose to_ do it anyway. In general, if I can, I tend to err on the side of a) having the `return` at the bottom, and b) having the `return` return something which fits on one line, possibly named, because I think it makes it easy to understand at a glance what's getting returned.

My reasoning with `function` vs arrows runs along the same lines - I want to be able to tell at a very quick glance that something's a function. To me, that matters more than the `this` question, because I feel like that actually doesn't come up much outside of classes (where I'm already thinking about it anyway).

But past functions having a name, if possible, so that stacktraces are easier to parse... the rest is just a matter of opinion, I guess. ¯\\_(ツ)_/¯